### PR TITLE
fix 404 urls in kustomization.md

### PR DESCRIPTION
**Pull request by npu21**
#17649

### CODE REVIEW

**Comment by npu21** [comment]
@tanjunchen I've closed https://github.com/kubernetes/website/pull/17734 for cla issue

File: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md [hunk 1]
```diff
@@ -808,14 +808,14 @@ deployment.apps "dev-my-nginx" deleted
...
-| vars                  | [][Var](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/var.go#L31)                       | Each entry is to capture text from one resource's field                            |
-| images                | [][Image](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/image/image.go#L23)                   | Each entry is to modify the name, tags and/or digest for one image without creating patches |
+| patchesJson6902       | [][Json6902](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/patchjson6902.go)             | Each entry in this list should resolve to a Kubernetes object and a Json Patch     |
+| vars                  | [][Var](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/var.go#L19)                       | Each entry is to capture text from one resource's field                            |
+| images                | [][Image](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/image.go)                   | Each entry is to modify the name, tags and/or digest for one image without creating patches |
```
Comment (sftim) [inline content/en/docs/tasks/manage-kubernetes-objects/kustomization.md]:
```suggestion
| images                | [][Image](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/image.go#L8)                   | Each entry is to modify the name, tags and/or digest for one image without creating patches |
```

**Comment by tengqm** [comment]
Links to source code lines in the master branch are fragile. They may change on a day-to-day basis. We should instead consider pinning the link to a specific branch.

File: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md [hunk 2]
```diff
@@ -808,14 +808,14 @@ deployment.apps "dev-my-nginx" deleted
...
 | resources             | []string                                                                                                     | each entry in this list must resolve to an existing resource configuration file    |
-| configmapGenerator    | [][ConfigMapArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/kustomization.go#L195)  | Each entry in this list generates a ConfigMap                                      |
-| secretGenerator       | [][SecretArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/kustomization.go#L201)     | Each entry in this list generates a Secret                                         |
-| generatorOptions      | [GeneratorOptions](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/kustomization.go#L239) | Modify behaviors of all ConfigMap and Secret generator                             |
+| configmapGenerator    | [][ConfigMapArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/kustomization.go#L99)  | Each entry in this list generates a ConfigMap                                      |
```
Comment (tengqm) [inline content/en/docs/tasks/manage-kubernetes-objects/kustomization.md]: To be more specific, please consider changing the link to something like
`https://github.com/kubernetes-sigs/kustomize/blob/release-kustomize-v4.0/api/types/kustomization.go#L99`
where the `master` is replaced by a tag `release-kustomize-v4.0`.
The revised link will not be a moving target then.

Comment (tanjunchen) [reply content/en/docs/tasks/manage-kubernetes-objects/kustomization.md]: @npu21 Maybe you can consider the view from tengqm .
Some files are archived and we should consider linking to them.
Thanks!

**Comment by Rajakavitha1** [comment]
Hi @npu21 !!! Thanks for the PR. Please updated the changes that @tengqm has suggested: https://github.com/kubernetes/website/pull/17793#discussion_r351052049

**Comment by npu21** [comment]
Hi, @Rajakavitha1 @tengqm @tanjunchen
good suggestion, adopted
Thanks!

**Comment by Rajakavitha1** [comment]
/lgtm
@tengqm could you please approve this PR!!

**Pull request merged**